tests/post-checkout.bats

```diff
@@ -2,62 +2,97 @@
 
-    assert_output --partial "INFO: The Buildkite config version (unresolved) is: d08b05df6f01cf4ec6e4c28ad94cedda76cc62e8"
-    assert_output --partial "commit d08b05df6f01cf4ec6e4c28ad94cedda76cc62e8"
-    assert_output --partial "Author: Viral B. Shah <ViralBShah@users.noreply.github.com>"
-    assert_output --partial "Date:   Thu Nov 11 18:48:11 2021 -0500"
+    test_known_checkout
 }
 
-@test "Unit test: first job" {
-    export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_EXTERNAL_REPO="https://github.com/JuliaLang/julia.git"
+@test "Embedded gitref" {
+    # Ouroboros mode engage
+    export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_REPO_URL="https://github.com/JuliaCI/external-buildkite-buildkite-plugin"
     export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_FOLDER="${TEMPORARY_DIRECTORY:?}/.buildkite"
-    export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_THIS_IS_FIRST_JOB="true"
-    export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_VERSION_FILE="${TEMPORARY_DIRECTORY:?}/buildkite.version"
+    export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_VERSION="test-anchor"
 
-    run rm -rf "${BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_VERSION_FILE:?}"
-    run bash -c "echo d08b05df6f01cf4ec6e4c28ad94cedda76cc62e8 > ${BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_VERSION_FILE:?}"
+    test_known_checkout
+}
+
+@test "File gitsha" {
+    export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_REPO_URL="https://github.com/JuliaCI/external-buildkite-buildkite-plugin"
+    export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_FOLDER="${TEMPORARY_DIRECTORY:?}/.buildkite"
+    export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_VERSION="./version"
 
-    test_post_checkout_hook
+    echo "27b8700184aa282a89e380449eb0b2a11b2c7d85" > ./version
+    test_known_checkout
 }
 
-@test "Unit test: subsequent job" {
-    export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_EXTERNAL_REPO="https://github.com/JuliaLang/julia.git"
+@test "File gitref" {
+    export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_REPO_URL="https://github.com/JuliaCI/external-buildkite-buildkite-plugin"
     export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_FOLDER="${TEMPORARY_DIRECTORY:?}/.buildkite"
-    export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_VERSION="d08b05df6f01cf4ec6e4c28ad94cedda76cc62e8"
+    export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_VERSION="./version"
 
-    test_post_checkout_hook
+    echo "test-anchor" > ./version
+    test_known_checkout
 }
+
+@test "buildkite-agent metadata override" {
+    export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_REPO_URL="https://github.com/JuliaCI/external-buildkite-buildkite-plugin"
+    export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_FOLDER="${TEMPORARY_DIRECTORY:?}/.buildkite"
+    export BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_VERSION="not a real version"
+
+    # Set ourselves an overriding meta version
+    buildkite-agent metadata set BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_VERSION "test-anchor"
+
+    assert [ $(buildkite-agent metadata get BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_VERSION) == "test-anchor" ]
+    test_known_checkout
+}
```

Comment:
Can you add a trailing newline to the end of the `tests/post-checkout.bats` file?